tests/test_cli.py

```diff
@@ -341,9 +473,81 @@ def test_changelog_should_call_functions(mocker, runner):
+                                           return_value='super changelog')
+
+    changelog(noop=True, unreleased=False)
+
+    mock_previous_version.assert_called_once_with('current')
+    mock_generate_changelog.assert_called_once_with('previous', 'current')
+    mock_markdown_changelog.assert_called_once_with('current', 'super changelog', header=False)
+
+
+def test_changelog_post_unreleased_no_token(mocker):
+    mocker.patch('semantic_release.cli.get_current_version', return_value='current')
+    mock_previous_version = mocker.patch('semantic_release.cli.get_previous_version',
+                                         return_value='previous')
+    mock_generate_changelog = mocker.patch('semantic_release.cli.generate_changelog',
+                                           return_value='super changelog')
+    mock_markdown_changelog = mocker.patch('semantic_release.cli.markdown_changelog',
+                                           return_value='super changelog')
+    mock_check_token = mocker.patch('semantic_release.cli.check_token',
+                                    return_value=False)
+
+    changelog(unreleased=True, post=True)
+
+    mock_previous_version.assert_called_once_with('current')
+    mock_generate_changelog.assert_called_once_with('current', None)
+    mock_markdown_changelog.assert_called_once_with('current', 'super changelog', header=False)
+    mock_check_token.assert_called_once_with()
+
+
+def test_changelog_post_complete(mocker):
+    mocker.patch('semantic_release.cli.get_current_version', return_value='current')
+    mock_previous_version = mocker.patch('semantic_release.cli.get_previous_version',
+                                         return_value='previous')
+    mock_generate_changelog = mocker.patch('semantic_release.cli.generate_changelog',
+                                           return_value='super changelog')
+    mock_markdown_changelog = mocker.patch('semantic_release.cli.markdown_changelog',
+                                           return_value='super md changelog')
+    mock_check_token = mocker.patch('semantic_release.cli.check_token',
+                                    return_value=True)
+    mock_get_owner_name = mocker.patch('semantic_release.cli.get_repository_owner_and_name',
+                                       return_value=('owner', 'name'))
+    mock_post_changelog = mocker.patch('semantic_release.cli.post_changelog')
+
+    changelog(unreleased=True, post=True)
+
+    mock_previous_version.assert_called_once_with('current')
+    mock_generate_changelog.assert_called_once_with('current', None)
+    mock_markdown_changelog.assert_any_call('current', 'super changelog', header=False)
+    mock_check_token.assert_called_once_with()
+    mock_get_owner_name.assert_called_once_with()
+    mock_post_changelog.assert_called_once_with('owner', 'name', 'current',
+                                                'super md changelog')
+
+
+def test_changelog_raises_exception(mocker):
```

Comment:
This test name should reflect what it checks, as in answer why do we expect an error here?